frontend/src/pages/LogsPage/SearchForm/utils.ts

```diff
@@ -0,0 +1,81 @@
+export type LogsSearchParam = {
+	key: string
+	operator: string
+	value: string
+	offsetStart: number
+}
+
+const SEPARATOR = ':'
+const DEFAULT_OPERATOR = '='
+const NAME_TEXT = 'text'
+const PARSE_REGEX =
+	/(\S+:'(?:[^'\\]|\\.)*')|(\S+:"(?:[^"\\]|\\.)*")|(-?"(?:[^"\\]|\\.)*")|(-?'(?:[^'\\]|\\.)*')|\S+|\S+:\S+|\s$/g
+
+export const parseLogsQuery = (query = ''): LogsSearchParam[] => {
+	if (query.indexOf(SEPARATOR) === -1) {
+		return [
+			{
+				key: NAME_TEXT,
+				operator: DEFAULT_OPERATOR,
+				value: query,
+				offsetStart: 0,
+			},
+		]
+	}
+
+	const terms = []
+	let match
+
+	while ((match = PARSE_REGEX.exec(query)) !== null) {
+		const term = match[0]
+
+		if (term.indexOf(SEPARATOR) > -1) {
+			const [key, value] = term.split(SEPARATOR)
+
+			terms.push({
+				key,
+				operator: DEFAULT_OPERATOR,
+				value: value?.replace(/^\"|\"$|^\'|\'$/g, ''), // strip quotes
```

Comment:
Is this still handling single quotes? If so, can we omit to match parity with the backend?

Reply:
Is backend only handling single or double quotes? I was thinking we would allow both on the FE and transform them before sending to the backend.

Reply:
The backend only handles double quotes. See #4365

Reply:
OK, thanks. Apologies, I could have looked that up myself 😄

Reply:
I'm going to look at this in a follow up PR.